.evg.yml

```
@@ -101,8 +101,7 @@ functions:
       working_dir: mongo-jdbc-driver
       script: |
           ${PREPARE_SHELL}
-          ./resources/run_adl.sh start
-          ./gradlew runDataLoader
+          ./resources/run_adl.sh start && ./gradlew runDataLoader
           ./gradlew clean integrationTest -x test
           ./resources/run_adl.sh stop
```

```suggestion
          ./resources/run_adl.sh start && 
          ./gradlew runDataLoader &&
          ./gradlew clean integrationTest -x test &&
          ./resources/run_adl.sh stop
```
Let's add the && for all the commands here, that way evergreen will see the final exit 1 if it occurs.

---

umh, so my understanding is that it won't execute the next command unless the previous one as an exit status != 1.
I do want to stop ADL even if integration tests fail. Adding one after runDataLoader is Ok. I wasn't sure what if Gradle was playing nice with the exit status, so I tested it and it does. Adding it right now.

---

There needs to be more logic in the script then to always run `./resources/run_adl.sh stop` but also return the exit code of 1 if one of the earlier steps fails.  
Something like (didn't test this myself):
```
./resources/run_adl.sh start && 
./gradlew runDataLoader &&
./gradlew clean integrationTest -x test
EXITCODE=$?
./resources/run_adl.sh stop
exit $EXITCODE
```

---

ok, let me try this with a "patch" using a broken dataloader and see what it does
Thank you for the help here :)

---

Tested the proposed solution, it works perfectly! Thank you @bucaojit :)